### PR TITLE
[enhancement] refactor test_run_to_run_stability_tests

### PR DESCRIPTION
**Pull request by icfaust**
### Description 
This refactors test_run_to_run_stability_tests into test_run_to_run_stability by using the centralized PATCHED_MODELS and other functions within the sklearnex/tests/_utils.py in order to centralize and automate discovery for testing coverage. This duplicates the current functionality such that GPU support can be easily added in a follow-on PR. This reduces the overall maintenance, and aligns with other sklearnex centralized testing.

Changes:
- Add proper calls for dispatching LogisiticRegression predict_proba and predict_log_proba
- Fix n_jobs support for predict_proba and predict_log_proba
- Add cpu checks which properly log fallbacks to sklearn
- Add allow_sklearn_fallback to n_jobs testing
- Change name of test_run_to_run_stability_tests to test_run_to_run_stability
- Create four tests, 1) estimators from patch map with defaults, 2) special instances with specifically set hyper parameters, 3) sparse testing, 4) specific instances that were previously tested in test_run_to_run_stability_tests
- Use get_dataframes_and_queues with pytest parametrize for eventual GPU and array_api testing
- Rewrite sklearnex/tests/_utils.py gen_dataset to be more generalized for use in test_run_to_run_stability
- Rework test_patching to work with the new gen_dataset
- shorten integer check in test_patching

Tasks

- [x] Compile
- [x] Correct issues discovered in public CI
- [x] Pass public CI
- [ ] Correct issues discovered in private CI
- [ ] Pass private CI

### CODE REVIEW

File: sklearnex/tests/test_n_jobs_support.py [hunk 1]
```diff
@@ -36,6 +36,7 @@
 X, Y = make_classification(n_samples=40, n_features=4, random_state=42)
 
 
+@pytest.mark.allow_sklearn_fallback
```
Comment (Alexsandruss) [inline sklearnex/tests/test_n_jobs_support.py:39]: Test of n_jobs support in sklearnex estimators is not expected to fall back on any patched method.

**Review by Alexsandruss — CHANGES_REQUESTED** [review]

File: sklearnex/tests/test_n_jobs_support.py (hunk 1 above)
Comment (icfaust) [reply sklearnex/tests/test_n_jobs_support.py:39]: And yet it does. LogisticRegression predict_proba and predict_log_proba silently fall back to sklearn in daal4py here: https://github.com/intel/scikit-learn-intelex/blob/main/daal4py/sklearn/linear_model/logistic_path.py#L820

File: daal4py/sklearn/linear_model/logistic_path.py [hunk 2]
```diff
@@ -951,9 +951,7 @@ def logistic_regression_path(
     )
 
 
-@control_n_jobs(
-    decorated_methods=["fit", "predict", "predict_proba", "predict_log_proba"]
-)
+@control_n_jobs(decorated_methods=["fit", "predict"])
```
Comment (Alexsandruss) [inline daal4py/sklearn/linear_model/logistic_path.py:954]:
```suggestion
@control_n_jobs(
    decorated_methods=["fit", "predict", "predict_proba", "predict_log_proba"]
)
```

File: sklearnex/linear_model/logistic_regression.py [hunk 3]
```diff
@@ -302,6 +300,14 @@ def _onedal_cpu_supported(self, method_name, *data):
...
                 f"sklearn.linear_model.{class_name}.{method_name}"
             )
+            patching_status.and_conditions(
+                [
+                    (
+                        method_name in ["fit", "predict", "score"],
+                        "only fit, predict, and score are supported by oneDAL on cpu",
+                    )
+                ]
+            )
```
Comment (Alexsandruss) [inline sklearnex/linear_model/logistic_regression.py:310]:
```suggestion
```

**Review by Alexsandruss — CHANGES_REQUESTED** [review]
Re-enabling of probabilities computation is required.

**Review by Alexsandruss — APPROVED** [review]

**Pull request merged**